agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1986,6 +1988,41 @@ bool backhaul_manager::handle_slave_backhaul_message(std::shared_ptr<sRadioInfo>
         memcpy(client_info.assoc_req, msg->association_frame(), client_info.asso_len);
         associated_clients[msg->client_mac()] = client_info;
 
+        // build 1905.1 message CMDU to send to the controller
+        if (!cmdu_tx.create(0, ieee1905_1::eMessageType::TOPOLOGY_NOTIFICATION_MESSAGE)) {
+            LOG(ERROR) << "cmdu creation of type TOPOLOGY_NOTIFICATION_MESSAGE, has failed";
+            return false;
+        }
+
+        auto client_association_event_tlv = cmdu_tx.addClass<wfa_map::tlvClientAssociationEvent>();
+        if (!client_association_event_tlv) {
+            LOG(ERROR) << "addClass tlvClientAssociationEvent failed";
+            return false;
+        }
+        client_association_event_tlv->client_mac() = msg->client_mac();
+        client_association_event_tlv->bssid()      = msg->bssid();
+        client_association_event_tlv->association_event() =
+            wfa_map::tlvClientAssociationEvent::CLIENT_HAS_JOINED_THE_BSS;
+
+        if (!is_prplmesh_controller) {
+            LOG(DEBUG) << "non-prlMesh, not adding ClientAssociationEvent VS TLV";
+        } else {
+            // Add vendor specific tlv
+            auto vs_tlv =
+                message_com::add_vs_tlv<beerocks_message::tlvVsClientAssociationEvent>(cmdu_tx);
+
+            if (!vs_tlv) {
+                LOG(ERROR) << "add_vs_tlv tlvVsClientAssociationEvent failed";
+                return false;
+            }
+
+            vs_tlv->mac()          = msg->client_mac();
```

Comment:
`tlvClientAssociationEvent` already contains the `client_mac` and the `bssid`, why do we pass it again in the VS TLV?

Reply:
The VS TLV is there to extend the standard TLV. Since there can be multiple tlvClientAssociationEvent TLVS in the same CMDU, we have to include the client mac and bssid also in the VS TLV so we can have a 1 to 1 mapping from VS TLV to non-VS tlv.

Reply:
This made me see that we will fail topology handling for non-prplMesh agents --> added a commit wihch fixes this - https://github.com/prplfoundation/prplMesh/pull/1296/commits/ea4c559192efeaf5cc276feaae473fc70d69e0e3

Reply:
But each `tlvClientAssociationEvent` will always be followed by `tlvVsClientAssociationEvent` (which should contain only the extended information).
Still don't understand why we need these fields in both TLVs.

Reply:
If we rely on the TLVs order, then we can remove these from the VS TLV. However, parsing will be much harder and less readable. I think that having the (small) duplication here is fine.